src/java/main/org/apache/zookeeper/server/PrepRequestProcessor.java

```diff
@@ -1,978 +0,0 @@
-            while (iter.hasPrevious()) {
-                ChangeRecord c = iter.previous();
-                if (c.zxid == zxid) {
-                    iter.remove();
-                    // Remove all outstanding changes for paths of this multi.
-                    // Previous records will be added back later.
-                    zks.outstandingChangesForPath.remove(c.path);
-                } else {
-                    break;
-                }
-            }
-
-            // we don't need to roll back any records because there is nothing left.
-            if (zks.outstandingChanges.isEmpty()) {
-                return;
-            }
-
-            long firstZxid = zks.outstandingChanges.get(0).zxid;
-
-            for (ChangeRecord c : pendingChangeRecords.values()) {
-                // Don't apply any prior change records less than firstZxid.
-                // Note that previous outstanding requests might have been removed
-                // once they are completed.
-                if (c.zxid < firstZxid) {
-                    continue;
-                }
-
-                // add previously existing records back.
-                zks.outstandingChangesForPath.put(c.path, c);
-            }
-        }
-    }
-
-    /**
-     * Grant or deny authorization to an operation on a node as a function of:
-     *
-     * @param zks: not used.
-     * @param acl:  set of ACLs for the node
-     * @param perm: the permission that the client is requesting
-     * @param ids:  the credentials supplied by the client
-     */
-    static void checkACL(ZooKeeperServer zks, List<ACL> acl, int perm,
-            List<Id> ids) throws KeeperException.NoAuthException {
-        if (skipACL) {
-            return;
-        }
-        if (LOG.isDebugEnabled()) {
-            LOG.debug("Permission requested: {} ", perm);
-            LOG.debug("ACLs for node: {}", acl);
-            LOG.debug("Client credentials: {}", ids);
-        }
-        if (acl == null || acl.size() == 0) {
-            return;
-        }
-        for (Id authId : ids) {
-            if (authId.getScheme().equals("super")) {
-                return;
-            }
-        }
-        for (ACL a : acl) {
-            Id id = a.getId();
-            if ((a.getPerms() & perm) != 0) {
-                if (id.getScheme().equals("world")
-                        && id.getId().equals("anyone")) {
-                    return;
-                }
-                AuthenticationProvider ap = ProviderRegistry.getProvider(id
-                        .getScheme());
-                if (ap != null) {
-                    for (Id authId : ids) {
-                        if (authId.getScheme().equals(id.getScheme())
-                                && ap.matches(authId.getId(), id.getId())) {
-                            return;
-                        }
-                    }
-                }
-            }
-        }
-        throw new KeeperException.NoAuthException();
-    }
```

Comment:
### ✅ No longer an issue: Deep, Nested Complexity
checkACL is no longer above the threshold for nested complexity depth